galileo/src/render/wgpu/mod.rs

```diff
@@ -295,15 +295,15 @@ impl WgpuRenderer {
 
     /// Creates a new renderer from the initialized wgpu structs.
     pub fn new_with_device_and_surface(
-        device: Arc<Device>,
+        device: &Device,
         surface: Arc<Surface<'static>>,
-        queue: Arc<Queue>,
+        queue: &Queue,
         config: SurfaceConfiguration,
     ) -> Self {
         let render_target = RenderTarget::Surface { surface, config };
         let mut renderer = Self {
-            device,
-            queue,
+            device: device.clone(),
+            queue: queue.clone(),
```

Comment:
It's usually better for the function signature to reflect what the function actually need and not hide cloning of stuff inside the body. In this case the function needs to have ownership of the `Device` and `Queue`, so it is better to give it owned values instead of refs and let the caller decide if cloning is needed or not.

For example, above in the `new_with_window` function current signature require additional cloning instead of just taking the values.

Reply:
Good point, looking at this again also reminded me that the surface parameter deserves the same treatment here.

I also noticed that raw window handle's traits are also implemented for Arc's of themselves, but that's probably worth a separate patch.